holoviews/core/util.py

```diff
@@ -163,6 +163,16 @@ class Config(param.ParameterizedFunction):
        Global default colormap for HeatMap elements. Prior to HoloViews
        1.14.0, the default value was the 'RdYlBu_r' colormap.""")
 
+    wikimedia_tile_source_replacement = param.String(default='OSM-with-warning', doc="""
+       If set to the special sentinal value of OSM-with-warning, replace
+       wikimedia tile source with OpenStreetMap (OSM) and warn. If set
+       explicitly with a suitable Tile source URL, apply replacement for
+       wikimedia tile source without warning.
+
+       This config parameter was introduced as wikimedia tile sources
+       can no longer be used outside the wikimedia domain, as of
+       April 2021.""")
```

Comment:
Damn GitHub reviews, thought I'd left this comment earlier but turns out I hadn't hit submit.

Personally I really don't think this config option is needed and don't think anyone should use it. Aliasing the Wikipedia tile source to refer to something other than the Wikipedia tile sources is weird and we're only doing it because it's used a bunch (and it's the default if you do `tiles=True` in hvplot). Any actual user should switch to the tile source they actually want, not do this aliasing.

Reply:
Seems like we need to change the hvplot default as well, then.

Reply:
Definitely.

Reply:
Fixed here: https://github.com/holoviz/hvplot/pull/610

Reply:
I am not sure I agree...the idea is that some people might be fine with replacing wikipedia with OSM throughout but don't want to keep seeing a warning. I would be fine making it into a bool if setting a different replacement than OSM seems weird...

Reply:
Or in other words, some people may be more annoyed by warnings appearing in old notebooks than the fact that the tile source that says it is wikipedia is now OSM...

Reply:
>(and may not want to update all their existing code/notebooks).

But to set the config option they would be doing this anyway, no? The difference in effort between doing it once at the top and doing a search and replace doesn't seem particularly high, and actually replacing the tile source is preferable since we do want to eventually remove Wikipedia (since keeping the inaccurate alias around forever does not seem like a great idea). 

Anyway no very strong opinions here but I've always been pretty inclined against config options to control backward compatibility options since they aren't very discoverable and I prefer nudging people to do the right thing rather than letting them live in their outdated world any longer and thereby forcing them to make edits to their code twice.

Reply:
> But to set the config option they would be doing this anyway, no?

Not necessarily: people can use the holoviews.rc to do it once only.

Happy to let @jbednar be the tie breaker!

Reply:
>Not necessarily: people can use the holoviews.rc to do it once only (or in other ways that are environment/sitewide)

I actually also really dislike the idea of rc files setting options like this because it leads to reproducibility nightmares.

Reply:
I don't mind using rc files to set backwards compatibility options, because by definition the code involved is already broken. Doing that makes it _more_ reproducible than it was, in the sense that having set it things will now mostly work.  So I'd say such compatibility options are just about the only reason to use rc files, and only as a temporary measure while knowing you eventually need to update the scattered source code.

Still, in this case, I don't think adding the configurability is very useful, because it only addresses one of the tile sources, with the others still broken. So I'm inclined to simply swap in for Wikipedia with a warning, only because it's the default in current hvplot versions, and otherwise for there simply to be a warning and require that people update their code to use a working tile source.

Reply:
> I actually also really dislike the idea of rc files setting options like this because it leads to reproducibility nightmares.

I feel the same way and only want rc files to be used when absolutely necessary (but it can be a useful escape hatch when needed as Jim says). 

> So I'm inclined to simply swap in for Wikipedia with a warning...

Ok, I'll get rid of the config option then.

Reply:
If you could also add the changelog then we can get this merged and released.